src/highlightjs/test/baselines/bad_inlinecomments.bicep

```diff
@@ -0,0 +1,3 @@
+var test = {
+  /* this doesn't work :( */

```

Comment:
Out of curiosity, why doesn't it work?

Reply:
Oh, it actually does for highlight.js, let me fix that up. These baselines were copy+pastes from the TextMate baselines.

Whitespace/comment handling is pretty painful in TextMate, because there's no straightforward way to recurse over two statements side-by-side - e.g. if you consider a regex for the opening of an object key-value:
`^(<whitespace_or_comment>)*(<ident>)(<whitespace_or_comment>)*:(<whitespace_or_comment>)*` where:
* `<whitespace_or_comment>` is something like `(\/\*.*\*\/|\t| )`
* `<ident>` is a regex for the key identifier (simplified as we actually support recursing with interpolated string)

Really, you want to recursively handle `<whitespace_or_comment>` so that you process and mark all the block comments correctly, but TM just doesn't give you a simple way to do this.

The approach I've taken for HighlightJs is to create a much less accurate, but way more permissive grammar - making it simpler to handle e.g. block comments as I'm being less strict about what goes where. It actually feels like a good tradeoff, and I'm wondering if it's work taking a similar approach for TextMate.

Reply:
BTW here's how you can check whether it's doing the right thing (hljs is correctly using the `hljs-comment` CSS class: https://github.com/Azure/bicep/blob/9fad9e7890ee63ca30a6860391e465a169b41bf1/src/highlightjs/test/baselines/bad_inlinecomments.html#L9

Reply:
Got it. This is good to know!